slatedb/src/compactor_state.rs

```diff
@@ -107,6 +107,32 @@ impl Display for CompactionSpec {
     }
 }
 
+/// Lifecycle status for a compaction.
+///
+/// This is currently tracked in-memory, but not in the .compactions file.
```

Comment:
I'm sure I could figure this out by digging deeper, but if we don't persist this how do we differentiate between orphaned/finished when we start a new compactor? Do we persist the compactor epoch with the compaction? (I don't think so?)

Reply:
We don't currently. The GC treats all compactions as active and simply takes the min. If the compaction is actually failed, it doesn't really matter from a GC perspective. We're only tracking the status in-memory so our trim() calls can work properly. Without it, if we have multiple active compaction jobs, the trim() method won't know which are safe to drop.